### DISCUSSION TITLE
[wish] help needed to install apps

### DISCUSSION THREAD

**Discussion by zeroheure** [trunkmaster/nextspace#378] [Ideas]
Nextspace is very easy and pleasant to use, but miss important softwares for a better usability. Actualy, very few Gnustep apps are available in Centos or Fedora — at least they are not easy to find.
I heavily suggest to add some help with end users in mind :

- some wrappers towards recommended softwares like Firefox, Thunderbird, emacs, LibO and so on.
- documentation to find available Gnustep apps, Flatpak, Snap and Appimages packages.

**Comment by trunkmaster** [comment]
Xavier, you are reading my thoughts. I’ll plan to add this in future releases. Although, my main goal for now is usability and stability. I don’t’ want to repeat general opensource style mistakes. My general concept is to provide less features in the sake of stability and speed.

Sergii Stoian

**Comment by zeroheure** [comment]
Totaly agree.

**Comment by zeroheure** [comment]
To remember : 

- as a basis, there is already some obsolete wrappers in GWorkspace (rather obsoletes)
- wrapperfactory.app creates wrappers for GWorkspace

I will test

**Comment by zeroheure** [comment]
For GNUSTEP packages, simple instructions like these from PikoPixel are a great help.

- http://twilightedge.com/gnustep/pikopixel/fedora_install_script.html

**Comment by BertrandDekoninck** [comment]
I've also a bunch of app wrappers in my RikIcons repo at https://github.com/BertrandDekoninck/RikIcons/tree/master/AppWrappers.

**Comment by nicktelindert** [comment]
@BertrandDekoninck Thank you! I was just in the process of making them myself. Maybe we can create a script to convert .desktop files to an app wrapper? A desktop file probably contains all the needed information. Maybe i look into it this week.

**Comment by nicktelindert** [comment]
I created a bash script to convert .desktop files to .app folders, it already works pretty well, but i consider it a pre-release.
Check it out: https://github.com/nicktelindert/GenerateXAppWrapper

**Comment by pcardona34** [comment]
Hello,

1) About GNUstep apps:

I am working on a way to install several well tested GNUstep apps one step from sources: the project is there: [apps-within-nextspace](https://github.com/pcardona34/apps-within-nextspace).

It is still a [work in progress](https://github.com/pcardona34/apps-within-nextspace/blob/main/Roadmap.md) . Already a few apps to test the concept. Until now, the dependencies are only set for debian 12 OS. Not yet available under Fedora like distros. But it is planned.

- How it works: it is a shell script and a hierarchy of Applications and Frameworks folders. A `desc.sh` script contains all the necessary source infos. Then, like in the BSD Ports, there is a batch of scripts to install dependencies, fetch the source, build and install the app...
- In the middle term, I would like to propose a GNUstep app to let the End user choose the apps to install/uninstall...

2) About Wrappers

It seems that this is now provided by NEXTSPACE: when You launch a foreign app, it is wrapped with a menu and an AppIcon.
The only problem is still the issue #454 for the users with a non-us layout keyboard.

**Comment by pcardona34** [comment]
As I  worked on my own desktop project in the while, I finally made an easy tool to manage and to install all kinds of GNUstep apps from a collection of Bash scripts and Dialog menus. As this work is quite stable, I plan to port the method to nextspace too.  I hope I will be back here soon to announce it. As the first tests within nextspace revealed some issues, I must polish it before I could publish the code. In the meanwhile, consider the above link as deprecated.
